### ISSUE TITLE
Feature request: page.on('dialog') support basic authentication

### ISSUE THREAD

**Issue by rasmusgadensgaard** [puppeteer/puppeteer#4588]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.10.0
* Platform / OS version: Windows Server 2016 Standard
* Node.js version: 8.12.0

It seems like page.on('dialog') does not support the basic auth dialog. 

I'm trying to achieve the following:

1. Open browser
2. Go to website
3. Take screenshot

Sometimes this website will ask for basic authentication, sometimes it will not. If I do the following: 
`await page.authenticate({username:"usr", password:"pwd",waitUntil: 'networkidle0'});` 

The page will hang if there is no login dialog. If I don't do the above, the page will hang if there is a login dialog. 

Basically two approaches would solve my issue: 

1: If I can detect if there is a basic login dialog
2. If I can do something like: await page.authenticate({user:usr:password:pwd, timeout: 30000})
  Basically allowing a timeout one the authentication.

**Comment by aslushnikov** [comment]
> The page will hang if there is no login dialog. If I don't do the above, the page will hang if there is a login dialog.

@rasmusgadensgaard any chance you can provide us with a script that we can run locally and reproduce this behavior?

**Comment by revivedsun** [comment]
> > The page will hang if there is no login dialog. If I don't do the above, the page will hang if there is a login dialog.

in headless mode, how to detect whether auth dialog has popped up? page.on('dialog') not work.

**Comment by aprilandjan** [comment]
I'm now facing almost the same problem as @revivedsun.I thought it was good to detect if the page is requiring basic authentication.But, If I already have the basic auth `username` and `password`, I could just always call `page.authenticate` at the beginning to ensure that the following requests are authenticated.
